RocketChat/Rocket.Chat
# fix: Old custom fields from migrated contacts are not ignored on update

## Proposed changes (including videos or screenshots)

 - On contact update, ignore custom fields that were previously in the contact, but are not registered in the workspace anymore (do not throw any error and do not change them);
 - On contact update, keep "legacy" custom fields (which are not registered in the workspace anymore) even if they are not specified in the update;

## Issue(s)


## Steps to test or reproduce

1. Create an omnichannel custom field (**Omnichannel > Custom fields**);
2. Create a contact using the custom field that has just been created;
3. Delete the custom field created in step 1;
4. Try updating the contact created in step 2 (using the UI or the `contacts.update` endpoint).

**Current behavior:** if the old custom field is provided in the update payload (this happens by default when using the UI to update the contact), then RC throws an error (caused by the unsupported custom field);
**Expected behavior:** no error should be thrown on update (even if the old custom field is provided). The old/legacy custom field should be kept unchanged. We should still throw an error in case extra custom fields (that are not in the contact before the update) are provided.

## Further comments

[SCI-193]

[SCI-193]: https://rocketchat.atlassian.net/browse/SCI-193?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
